Created endpoint to return movies that have not been rated and to send all required info to the page for movie rating

rrhabib opened the pull request:
Created embedded unit tests for data to be returned in application.py endpoints
Commented step-by-step process within the movie_options endpoint
Updated javascript file to take in data from endpoint and print it out onto HTML
Udpated basic HTML layout

rrhabib committed:
pushing for dployment to test front end connections in movie display

rrhabib committed:
Created database structure for movie_rates_groups and built on movie results endpoint

rrhabib committed:
Created the ednpoint to output a movie to be rated and pass along all critical information for the movie rating process

rrhabib committed:
Fixed merge issues that Github magically created to make my life hard

AdamSyed commented on app.js:
```diff
@@ -47,8 +47,28 @@ async function create_user(){
 async function returnMovie() {
     //this function will return the movie
-    //const api_endpoint = 'rating';
+    const RETURN_MOVIE_URL = 'https://cors-anywhere.herokuapp.com/http://moviefinder.us-east-1.elasticbeanstalk.com/rating/';   
```
I think there is value in being consistent with how we handle the endpoint links. Since we use the global variable and CONCAT for the other functions I think it might be good to do the same here. Thoughts? Did you do this for a particular reason?

rrhabib replied:
I decided to use a larger constant for the Movie URL in the returnMovie function because the concatenation was giving me some issues in testing and I wanted to make sure that this wasn't an additional step where the method could fail.  Although it is inconsistent, I believe that it would be good to keep it as is for now until we can spend more time on this method in Sprint 2 and become more comfortable with our approach in this method.

AdamSyed commented on application.py:
```diff
@@ -117,28 +120,30 @@ class Meta: #all the variables you want to see
+    userID = db.Column (db.Integer, db.ForeignKey(User.userID),primary_key=True)
+    isLiked = db.Column (db.Boolean, nullable = False)
+
+    #user = db.relationship(User, backref=db.backref("ratings", cascade="all, delete-orphan"))
```
If this is no longer used, I think we should delete this to avoid any confusion. Thoughts?

rrhabib replied:
I'm okay with deleting it if that makes the code cleaner and clearer.  This piece of code was originally left in in case it would be useful when creating the rest of the relationship endpoints in Flask that will be made in Sprint 2.  However, as I reflect on this, I agree that it may be best to remove it since we will not have relationship that will use this different approach.

rrhabib replied:
Have deleted the line. Went through the code and found a few more similar pieces of redundant code.  They have also been removed.  Thanks for pointing this out

AdamSyed reviewed (commented):
Overall, it looks great. Nice work with the backend endpoint. I have a few comments to clean it up. Let me know what your thoughts are.

rrhabib committed:
Updated application.py to remove vestigial code 

Noticed a number of unnecessarily commented code that Adam pointed out in the review of my pull request

AdamSyed committed:
Merge branch 'main' into rafiq
